core/transaction.c

```diff
@@ -234,11 +234,52 @@ void transaction_handle_response(lwm2m_context_t * contextP,
                 // So we resend transaction that were denied for authentication reason.
                 if (message->code != COAP_401_UNAUTHORIZED || transacP->retrans_counter >= COAP_MAX_RETRANSMIT)
```

Comment:
Hi David,
shortly comment: tricky I know ;-) Transaction is work in process coming next week by my colleague. Splitting is OK but I will review it on Monday.